sympy/physics/mechanics/_system.py

```diff
@@ -0,0 +1,937 @@
+"""Implementation of system for book-keeping all objects defining a model.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from sympy.core.backend import Basic, ImmutableMatrix as Matrix
+from sympy.core.containers import OrderedSet
+from sympy.utilities.iterables import iterable
+from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
+from sympy.physics.mechanics.functions import _validate_coordinates, Lagrangian
+from sympy.physics.mechanics.body_base import BodyBase
+from sympy.physics.mechanics.particle import Particle
+from sympy.physics.mechanics.loads import Force, Torque, _parse_load, gravity
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.mechanics.method import _Methods
+from sympy.physics.mechanics.kane import KanesMethod
+from sympy.physics.mechanics.lagrange import LagrangesMethod
+from functools import wraps
+
+__all__ = ['System']
+
+
+def _reset_eom_method(method):
+    """Decorator to reset the eom_method if a property is changed."""
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        self._eom_method = None
+        return method(self, *args, **kwargs)
+
+    return wrapper
+
+
+class System(_Methods):
+    """Class to define a system and form the equations of motion.
+
+    Explanation
+    ===========
+
+    A ``System`` instance stores the different objects associated with a model,
+    including bodies, joints, constraints, and other relevant information. With
+    all the relationships between components defined, the ``System`` can be used
+    to form the equations of motion using a backend, such as ``KanesMethod``.
+    The ``System`` has been designed to be compatible with third-party
+    libraries for greater flexibility and integration with other tools.
+
+    Attributes
+    ==========
+
+    origin : Point
+        Global origin of the system.
+    frame : ReferenceFrame
+        Inertial reference frame of the system.
+    x : Vector
+        Unit vector in the x direction of the inertial reference frame.
+    y : Vector
+        Unit vector in the y direction of the inertial reference frame.
+    z : Vector
+        Unit vector in the z direction of the inertial reference frame.
+    q : Matrix
+        Matrix of all the generalized coordinates.
+    u : Matrix
+        Matrix of all the generalized speeds.
+    q_ind : Matrix
+        Matrix of the independent generalized coordinates.
+    q_dep : Matrix
+        Matrix of the dependent generalized coordinates.
+    u_ind : Matrix
+        Matrix of the independent generalized speeds.
+    u_dep : Matrix
+        Matrix of the dependent generalized speeds.
+    kdes : Matrix
+        Matrix of the kinematic differential equations.
+    bodies : tuple of BodyBase subclasses
+        Tuple of all bodies that make up the system.
+    joints : tuple of Joint
```

Comment:
Is stating the types like this better than `tuple[Joint]`? Again, asking for my own clarification.

Reply:
It is rendered as text, so this way there is a little more freedom and maybe readability, otherwise is should also be `tuple[Joint, ...]`. Can change it if that is preferred.